src/ynEIGEN/TokenStakingNode.sol

```diff
@@ -64,35 +78,32 @@ contract LSDStakingNode is ILSDStakingNode, Initializable, ReentrancyGuardUpgrad
     //--------------------------------------------------------------------------------------
 
     /**
-     * @notice Deposits multiple assets into their respective strategies on Eigenlayer by retrieving them from ynLSD.
+     * @notice Deposits multiple assets into their respective strategies on Eigenlayer by retrieving them from tokenStakingNodesManager.
      * @dev Iterates through the provided arrays of assets and amounts, depositing each into its corresponding strategy.
      * @param assets An array of IERC20 tokens to be deposited.
      * @param amounts An array of amounts corresponding to each asset to be deposited.
      */
     function depositAssetsToEigenlayer(
         IERC20[] memory assets,
-        uint256[] memory amounts
-    )
-        external
-        nonReentrant
-        onlyLSDRestakingManager
-    {
-        IStrategyManager strategyManager = ynLSD.strategyManager();
+        uint256[] memory amounts,
+        IStrategy[] memory strategies
+    ) external nonReentrant onlyStrategyManager {
+        IStrategyManager strategyManager = tokenStakingNodesManager
+            .strategyManager();
 
         for (uint256 i = 0; i < assets.length; i++) {
```

Comment:
will likely get flagged for gas optimization: for (assets.length) -> uint256 assetLength = assets.length;